### PR TITLE
Initial configuration to allow setting up OTP secrets

### PR DESCRIPTION
**Pull request by cg505**
This is the first PR that must be merged for OTP sudo. It sets up the puppetserver/desktops to be ready to create and store secrets.

### CODE REVIEW

File: modules/ocf_puppet/files/write-otp-config [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage: write-otp-config <username> <yubikey public ID>"
+    exit 1
+fi
+
+username="$1"
+yubikeyID="$2"
+
+if grep -q "$username" /opt/puppet/shares/private/otp/yubikeys; then
```
Comment (fydai) [inline modules/ocf_puppet/files/write-otp-config]: grep -q "^$username"

Comment (dkess) [reply modules/ocf_puppet/files/write-otp-config]: actually should be `grep -q '^$username:`

File: modules/ocf_puppet/files/write-otp-config [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+username="$1"
+yubikeyID="$2"
+
+if grep -q "^$username:" /opt/puppet/shares/private/otp/yubikeys; then
+    echo "$username already registered. Appending yubikey, not reconfiguring google authenticator."
+    sed -i "/^$username:/ s/$/:$yubikeyID/" /opt/puppet/shares/private/otp/yubikeys
+    exit 0
+fi
+
+echo "$username:$yubikeyID" >> /opt/puppet/shares/private/otp/yubikeys
```
Comment (fydai) [inline modules/ocf_puppet/files/write-otp-config]: does this add you twice? if so, shouldn't this be in a else block

Comment (cg505) [reply modules/ocf_puppet/files/write-otp-config]: see line 16

File: modules/ocf_puppet/files/write-otp-config [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+set -euo pipefail
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage: write-otp-config <username> <yubikey public ID>"
+    exit 1
+fi
+
+username="$1"
+yubikeyID="$2"
```
Comment (kpengboy) [inline modules/ocf_puppet/files/write-otp-config]: Nit: in this repository, it seems the established convention is to use lowercase_with_underscores

File: modules/ocf_puppet/files/write-otp-config [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
```
Comment (kpengboy) [inline modules/ocf_puppet/files/write-otp-config]: Inform me, how is this script intended to be run? Is this supposed to be run by the user or some other job?

**Comment by cg505** [comment]
btw, with feedback from fydai, a lot of this script is pending rewrite. This is not in a mergable state right now

**Comment by cg505** [comment]
I'm thinking that I'm going to move the script to utils and expect people to call it directly on lightning. We don't really need this amount of tooling, it's just all obscuring what's actually happening.

**Comment by cg505** [comment]
okay, I removed the script, drastically simplified it, and moved it to ocf/utils#140

**Comment by cg505** [comment]
would appreciate some new reviews since all the reviews are super old now (although the code hasn't really changed).

**Pull request merged**